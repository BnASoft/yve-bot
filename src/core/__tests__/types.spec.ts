import YveBot from '..';

test('extend module type', () => {
  YveBot.types.extend('CustomString', 'String', {
    transform: async () => 'transform2',
    validators: [{ warning: 'validator 2' }],
  });

  expect((YveBot.types as any).CustomString).toMatchSnapshot();
});

test('define module type with short mode', () => {
  YveBot.types.define('compact', {
    customProp: '123',
    transform: () => 'value',
    validators: [
      { number: true, warning: 'Message 1' },
      { required: true, warning: 'Message 2' },
    ],
  });

  expect((YveBot.types as any).compact).toMatchSnapshot();
});
