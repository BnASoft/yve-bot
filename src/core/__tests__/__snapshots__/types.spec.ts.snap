// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`define module type with short mode 1`] = `
Object {
  "customProp": "123",
  "executors": Array [
    Object {
      "transform": [Function],
      "validators": Array [
        Object {
          "number": true,
          "warning": "Message 1",
        },
        Object {
          "required": true,
          "warning": "Message 2",
        },
      ],
    },
  ],
}
`;

exports[`extend module type 1`] = `
Object {
  "executors": Array [
    Object {
      "transform": [Function],
      "validators": Array [],
    },
    Object {
      "transform": [Function],
      "validators": Array [
        Object {
          "warning": "validator 2",
        },
      ],
    },
  ],
}
`;
